es/es-repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobContainer.java

```diff
@@ -95,14 +94,11 @@ public InputStream readBlob(String blobName) throws IOException {
      */
     @Override
     public void writeBlob(String blobName, InputStream inputStream, long blobSize, boolean failIfAlreadyExists) throws IOException {
-        SocketAccess.doPrivilegedIOException(() -> {
```

Comment:
not really sure about this SocketAccess thing, but would running this outside the "privileged io" block change semantics and behaviour in any way?

Reply:
It would cause issues if we enabled the security manager. Other than the implied permission checks there shouldn't be any semantic change.